---
title: "A_GTP_exploring_phenotype_data"
output:
  html_notebook:
    toc: yes
    toc_float: yes
date: "Jan. 21, 2020"
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, echo=FALSE}
.libPaths( c( "/opt/R/3.6.2/lib/R/library" , .libPaths() ) )
.libPaths()
```

```{r}
library(tidyverse)

library(readxl)
require(data.table)
```


## load datasets

```{r}
# phenotype data
GTP_pheno1 <- read_delim("/data/humgen/GTP_genomics/PLINK/input/pheno/gtp_screen_pheno.txt",
                         col_names = TRUE, delim = "\t");
GTP_pheno2 <- read_delim("/data/humgen/GTP_genomics/PLINK/input/pheno/Masterdataset_2-2-15_cleaned_reduced_For_Machine_Learning_5-25-2016.txt",
                         col_names = TRUE, delim = "\t");
```

```{r}
GTP_pheno3 <- read_excel("/data/humgen/GTP_genomics/PLINK/input/pheno/ResslerData_cleaned_reduced_For Machine Learning_withSIDs.xlsx", sheet = 1) %>% 
  print() 
```
```{r}
GTP_pheno3 %>% 
#  count(sex)
#  count(PTSD_diagnosis)
  subset(select=c("sex","PTSD_diagnosis")) %>% table()

```


```{r}
# install necessary packages
#install.packages("arsenal",  method = "curl", lib = "/PHShome/bm363/R/x86_64-pc-linux-gnu-library/3.6")
#install.packages("Amisc",  method = "curl", lib = "/PHShome/bm363/R/x86_64-pc-linux-gnu-library/3.6")
#install.packages("table1",  method = "curl", lib = "/PHShome/bm363/R/x86_64-pc-linux-gnu-library/3.6")

# clean up data for demo table
tab_data1 <- GTP_pheno3 %>% 
  mutate(PTSD_diagnosis = na_if(na_if(PTSD_diagnosis,"2092"),"7011")) %>% 
  mutate(PTSD_diagnosis = factor(PTSD_diagnosis, labels = c("Controls","Cases"))) %>% 
  filter(!is.na(PTSD_diagnosis))

# we can use this package if we need significance tests conducted
library(arsenal)

sum_tab1 <- tableby(PTSD_diagnosis ~ PSStotal + BDItotalscore + as.factor(BDI_CAT) + age + as.factor(sex) + education + CTQTOT + as.factor(race_ethnic), data=tab_data1, test=T)
summary(sum_tab1)


# a better lookong descriptive tables can be generated with the following package
library(table1)
label(tab_data1$PTSD_diagnosis) <- "PTSD Diagnosis"
label(tab_data1$age) <- "Age"
label(tab_data1$sex) <- "Sex"
label(tab_data1$race_ethnic) <- "Race/Ethnicity"
label(tab_data1$CTQTOT) <- "Childhood trauma score"

sum_tab2 <- table1(~ PSStotal + BDItotalscore + age + as.factor(sex) + education + CTQTOT + as.factor(race_ethnic) | PTSD_diagnosis, 
                   data = tab_data1)#, overall = FALSE)
sum_tab2
```

What is a better trauma exposure measure we should use?
Do we have CRP and other molecular markers measured?
Do we have BMI and other anthropometric measurements?



```{r}
GTP_pheno4 <- GTP_pheno3 %>% 
#  filter(!is.na(PTSD_status)) %>% 
#  filter(PTSD_status<125) %>% 
  subset(select=c("PTSD_diagnosis", "CTQTOT","education")) %>% 
  mutate(CA4 = cut(CTQTOT, breaks = c(-Inf,40,60,80,Inf), right=T)) %>%
  mutate(EA4 = cut(education, breaks = c(-Inf,1,3,4,Inf), right=T)) 

GTP_pheno4 %>% print()
#, labels=c("low","midlow","midhigh","high")
```

```{r}
x=table(GTP_pheno4[,c("PTSD_diagnosis","CA4")])
y <- x[c(2,1),]
barplot(y)

#par(mfrow=c(1, 1), mar=c(5, 5, 5, 10), xpd=TRUE)
barplot(prop.table(y, margin = 2), beside=F, horiz=F,
        col=c("red","steelblue"),
        legend = rownames(y), legend.text = TRUE,
        args.legend = list(x = "topright", bty = "n", inset=c(-0.25,0)),
        cex.lab=1.5,
        main = "Conditional probability of PTSD diagnosis",
        ylab = "Proportion with PTSD",
        xlab = "Education level")#,
        #names.arg = c("H.S. Diploma \n or less","A.A. Degree \n(2 yrs)","Bachelor's \n Degree (4 yrs)","Master's \n Degree or more"))

```


## Genotype data 

```{bash}
# Number of variants and samples
wc -l /data/humgen/GTP_genomics/PLINK/input/PGC_GTP_June2017/GTPC/qc/pts_gtpc_mix_am-qc.{bim,fam}

# an alternative geno data file location (befoe QC with) with slightly more samples and variants
wc -l /data/humgen/GTP_genomics/PLINK/input/PGC_GTP_June2017/GTPC/starting_data/Adam_final_GTP_pheno.{bim,fam}
```


```{r}
GTP_fam1 <- read_delim("/data/humgen/GTP_genomics/PLINK/input/PGC_GTP_June2017/GTPC/qc/pts_gtpc_mix_am-qc.fam",
                         col_names = F, delim = "\t") %>% 
  rename_all(funs(c("FID","IID","p1","p2","gend","phen"))) %>% 
  mutate(ID = str_replace(IID,"PTSD_",""))

GTP_pheno4 <- GTP_pheno3 %>% 
  mutate(sid = as.character(sid)) %>% 
  inner_join(GTP_fam1, by=c("sid"="ID")) 

write.table(GTP_pheno4, "/data/humgen/burook/GTP_analysis/GTP_pheno1.txt", row.names = FALSE, col.names = TRUE, quote = FALSE, sep = "\t")

```


## Ancestry estimation and confirmation

Prepare a file with self-identified race.
```{r}
GTP_race2 <- GTP_pheno4 %>% 
  subset(select=c("IID","race_ethnic")) %>% 
  rename(Ethnicity_race=race_ethnic)

write.table(GTP_race2, "/data/humgen/burook/GTP_analysis/GTP_race2.txt", row.names = FALSE, col.names = TRUE, quote = FALSE, sep = "\t")

```

```{bash}
cd /data/humgen/burook/GTP_analysis/
Rscript /data/humgen/burook/AURORA/notebooks/ancestry_prediction_svmandtree.R /data/humgen/GTP_genomics/PLINK/input/PGC_GTP_June2017/GTPC/qc/pts_gtpc_mix_am-qc /data/humgen/burook/GTP_analysis/GTP_race2.txt

```


```{r}
target_race <- read_delim("/data/humgen/burook/GTP_analysis/GTP_race2.txt", col_names = TRUE, delim = "\t")

 t1 %>% 
   left_join(target_race, by=c("Sample"="IID")) %>% 
   filter(!is.na(Ethnicity_race.y)) %>% 
   subset(select=c("Ethnicity_race.y","predtree")) %>% 
   table() %>% 
   print()

```

## Computing PRS

```{bash}
cd /data/humgen/burook/GTP_analysis/gtp_ptsd/

/PHShome/bm363/bin/plink2 \
  --bfile /data/humgen/GTP_genomics/PLINK/input/PGC_GTP_June2017/GTPC/qc/pts_gtpc_mix_am-qc \
  --maf 0.01 \
  --geno 0.01 \
  --mind 0.1 \
  --hwe 0.001 \
  --make-bed \
  --out tmp_gtp1

```


```{bash}
# with freeze-2 PGC-PTSD summ stats
R --file=/PHShome/bm363/bin/PRSice_v1.25/PRSice_v1.25.R -q --args  \
  plink /PHShome/bm363/bin/plink2 \
  base /data/humgen/burook/sum_stats/pts_aam_freeze2_overall.results \
  target /data/humgen/burook/GTP_analysis/gtp_ptsd/tmp_gtp1 \
  pheno.file /data/humgen/burook/GTP_analysis/GTP_race2.txt \
  binary.target T \
  covariates C1,C2,C3,C4,C5,C6,C7,C8,C9,C10 \
  clump.snps T \
  prune.snps F \
  quantiles T \
  figname gtp_ptsd \
  wd /data/humgen/burook/GTP_analysis/gtp_ptsd/
```



