---
title: "D_PRS_computation_and_analysis"
output:
  html_notebook:
    toc: yes
    toc_float: yes
date: "May 5, 2020"
---


```{r}
.libPaths( c( "/opt/R/3.6.2/lib/R/library" , .libPaths() ) )
.libPaths()
```

```{r}
library(tidyverse)
library(readxl)
```


# Computing PRS with  PRSice

## PTSD-PRS

Prepare PTSD pheno file for PRSice

```{r}
# fam_file <- read_delim("/data/humgen/burook/data_raw/sb/pts_sb1_mix_bm-qc1.hg19.ch.fl.bg.fam", col_names = FALSE, delim = " ");
# PTSD_clinical <- read_delim("/data/humgen/burook/data_raw/sb/PTSD_clinical.txt", col_names = TRUE, delim = "\t");
# PTSD_status2 <- PTSD_clinical %>% 
#   mutate(status = "-") %>% 
#   mutate(status = replace(status, PTSD_status=="Negative", 0)) %>% 
#   mutate(status = replace(status, PTSD_status=="Positive", 1)) %>% 
#   filter(status!="-") %>% 
#   subset(select=c("ID","status")) %>% 
#   print()
# write.table(PTSD_status2, "/data/humgen/burook/SysBio_analysis/sb_ptsd/pheno_file_status2", row.names = FALSE, col.names = FALSE, quote = FALSE)


# all samples
t1 <- read_delim("/data/humgen/burook/SysBio_analysis/table_pred", col_names = TRUE, delim = " ")
PTSD_status2 <- read_delim("/data/humgen/burook/data_raw/sb/PTSD_clinical.txt", col_names = TRUE, delim = "\t") %>%
  mutate(ID=as.character(ID)) %>% 
  inner_join(t1, by=c("ID"="Sample")) %>% 
  filter(PTSD_status=="Negative" | PTSD_status=="Positive") %>% 
  mutate(PTSD_status = factor(PTSD_status, labels =c(0,1))) %>% 
  subset(select=c("ID","PTSD_status")) %>% 
  print() 

write.table(PTSD_status2, "/data/humgen/burook/SysBio_analysis/sb_ptsd/pheno_file_status2", row.names = FALSE, col.names = FALSE, quote = FALSE)

# AFR ancestry samples
PTSD_status2_afr <- read_delim("/data/humgen/burook/data_raw/sb/PTSD_clinical.txt", col_names = TRUE, delim = "\t") %>%
  mutate(ID=as.character(ID)) %>% 
  inner_join(t1, by=c("ID"="Sample")) %>% 
  filter(predtree=='AFR') %>% 
  filter(PTSD_status=="Negative" | PTSD_status=="Positive") %>% 
  mutate(PTSD_status = factor(PTSD_status, labels =c(0,1))) %>% 
  subset(select=c("ID","PTSD_status")) %>% 
  print() 

write.table(PTSD_status2_afr, "/data/humgen/burook/SysBio_analysis/sb_ptsd_afr/pheno_file_status2_afr", row.names = FALSE, col.names = FALSE, quote = FALSE)

# AFR ancestry samples
PTSD_status2_nonafr <- read_delim("/data/humgen/burook/data_raw/sb/PTSD_clinical.txt", col_names = TRUE, delim = "\t") %>%
  mutate(ID=as.character(ID)) %>% 
  inner_join(t1, by=c("ID"="Sample")) %>% 
  filter(predtree!='AFR') %>% 
  filter(PTSD_status=="Negative" | PTSD_status=="Positive") %>% 
  mutate(PTSD_status = factor(PTSD_status, labels =c(0,1))) %>% 
  subset(select=c("ID","PTSD_status")) %>% 
  print() 

write.table(PTSD_status2_nonafr, "/data/humgen/burook/SysBio_analysis/sb_ptsd_nonafr/pheno_file_status2_nonafr", row.names = FALSE, col.names = FALSE, quote = FALSE)

```


```{bash}
cd /data/humgen/burook/SysBio_analysis/sb_ptsd/

/PHShome/bm363/bin/plink2 \
  --bfile /data/humgen/burook/data_imputed/sb/cobg_dir_genome_wide/pts_sb1_mix_bm-qc1.hg19.ch.fl.bgn \
  --maf 0.01 \
  --geno 0.01 \
  --mind 0.1 \
  --hwe 0.001 \
  --make-bed \
  --out tmp_sb1
```

```{bash}
# format the sum data for PRSice
head /data/humgen/burook/sum_stats/pts_eur_freeze2_overall.results
sort -k 2,2 -r -u pts_eur_freeze2_overall.results > pts_eur_freeze2_overall.results.exDupVar
```

Run PRSice with freeze-2 GWAS data
```{bash}
# with freeze-2 PGC-PTSD summ stats
R --file=/PHShome/bm363/bin/PRSice_v1.25/PRSice_v1.25.R -q --args  \
  plink /PHShome/bm363/bin/plink2 \
  base /data/humgen/burook/sum_stats/pts_eur_freeze2_overall.results.exDupVar \
  target /data/humgen/burook/SysBio_analysis/sb_ptsd/tmp_sb1 \
  pheno.file /data/humgen/burook/SysBio_analysis/sb_ptsd/pheno_file_status2 \
  binary.target T \
  covariates C1,C2,C3,C4,C5,C6,C7,C8,C9,C10 \
  clump.snps T \
  prune.snps F \
  quantiles T \
  figname sb_ptsd \
  wd /data/humgen/burook/SysBio_analysis/sb_ptsd/
```

```{bash}
# with freeze-2 PGC-PTSD summ stats
R --file=/PHShome/bm363/bin/PRSice_v1.25/PRSice_v1.25.R -q --args  \
  plink /PHShome/bm363/bin/plink2 \
  base /data/humgen/burook/sum_stats/pts_eur_freeze2_overall.results.exDupVar \
  target /data/humgen/burook/SysBio_analysis/sb_ptsd/tmp_sb1 \
  pheno.file /data/humgen/burook/SysBio_analysis/sb_ptsd_nonafr/pheno_file_status2_nonafr \
  binary.target T \
  covariates C1,C2,C3,C4,C5,C6,C7,C8,C9,C10 \
  clump.snps T \
  prune.snps F \
  quantiles T \
  figname sb_ptsd_nonafr \
  wd /data/humgen/burook/SysBio_analysis/sb_ptsd_nonafr/
```

```{bash}
# with freeze-2 PGC-PTSD summ stats
R --file=/PHShome/bm363/bin/PRSice_v1.25/PRSice_v1.25.R -q --args  \
  plink /PHShome/bm363/bin/plink2 \
  base /data/humgen/burook/sum_stats/pts_aam_freeze2_overall.results \
  target /data/humgen/burook/SysBio_analysis/sb_ptsd/tmp_sb1 \
  pheno.file /data/humgen/burook/SysBio_analysis/sb_ptsd_afr/pheno_file_status2_afr \
  binary.target T \
  covariates C1,C2,C3,C4,C5,C6,C7,C8,C9,C10 \
  clump.snps T \
  prune.snps F \
  quantiles T \
  figname sb_ptsd_afr_aam \
  wd /data/humgen/burook/SysBio_analysis/sb_ptsd_afr/
```
 
 
## Analysis of the constructed PRS 
 
```{r}
# let's load necessary files for performance analysis and plots
sb_pheno1 <- read_delim("/data/humgen/burook/data_raw/sb/PTSD_clinical.txt", col_names = TRUE, delim = "\t") %>%
  mutate(ID=as.character(ID)) 
pcs1 = read.table("/data/humgen/burook/SysBio_analysis/sb_ptsd_nonafr/ANCESTRY_INFORMATIVE_DIMENSIONS.mds", header = TRUE) %>% 
   mutate(IID=as.character(IID))
prs1 = read.table("/data/humgen/burook/SysBio_analysis/sb_ptsd_nonafr/sb_ptsd_nonafr_SCORES_AT_BEST-FIT-PRS.txt", header = TRUE) %>% 
  mutate(IID=as.character(IID)) 
prs1[,2]=scale(prs1[,2],center=T,scale=T); colnames(prs1)=c("IID","prs");
# also let's load the file containing predicted ancestry labels
t1 <- read_delim("/data/humgen/burook/SysBio_analysis/table_pred", col_names = TRUE, delim = " ")


# 
# pcs1 = read.table("/data/humgen/burook/SysBio_analysis/sb_ptsd_afr/ANCESTRY_INFORMATIVE_DIMENSIONS.mds", header = TRUE) %>% 
#    mutate(IID=as.character(IID))
# prs1 = read.table("/data/humgen/burook/SysBio_analysis/sb_ptsd_afr/sb_ptsd_afr_aam_SCORES_AT_BEST-FIT-PRS.txt", header = TRUE) %>% 
#   mutate(IID=as.character(IID))
# prs1[,2]=scale(prs1[,2],center=T,scale=T); colnames(prs1)=c("IID","prs");
# 


PTSD_all_data <- t1 %>% 
  filter(is.na(SuperPopulation)) %>% 
  mutate(IID=Sample) %>% 
  full_join(sb_pheno1, by=c("Sample"="ID")) %>%
  full_join(pcs1, by=("IID")) %>% 
  full_join(prs1, by=("IID")) %>% 
  filter(!is.na(prs)) %>% 
#  filter(predtree=='EUR') %>% 
  print()

```


```{r}
require(fmsb) # for NagelkerkeR2 calculation

predictors1 <- c("prs")
covariates <- c("C1","C2","C3","C4","C5","C6","C7","C8","C9","C10")  # c(,"DemoAge","DemoGend")

model.full <- glm(as.factor(PTSD_status) ~., family="binomial", data = PTSD_all_data[,c("PTSD_status", predictors1, covariates)])
model.null <-  glm(as.factor(PTSD_status) ~., family="binomial", data = PTSD_all_data[,c("PTSD_status",covariates)])

paste("R^2 = ", 
100*(NagelkerkeR2(model.full)$R2 - NagelkerkeR2(model.null)$R2),  # Nagelkerke R^2
", P = ",
summary(model.full)$coefficients[2,4]  ) # p-value only when there is only one predictor variable

```

```{r}
source('/data/humgen/burook/ptsd_multimodal_prediction_stage1/R/quantile_plot_and_stats.R')
#pdf("//.../quantiles_oddsRatio_ptsd_prs1.pdf", width=5.5, height=5)
plot1=quantile_plot_and_stats(prs = prs1, 
                              pcs = pcs1, 
                              pheno.data.all = PTSD_all_data, 
                              covariates = c("C1","C2","C3","C4","C5"),
                              num.quantiles = 4, binary.pheno = T, pheno.name = "PTSD_status")
#                              num.quantiles = 4, binary.pheno = F, pheno.name = "CAPSTOT_cur")

plot1

#dev.off()
#plot1
```

Distribution of the PTSD-PRS 
```{r}
plot1 <- PTSD_all_data %>% 
  filter(!is.na(prs)) %>% 
ggplot(aes(x=prs)) + 
  geom_histogram(aes(y =..density..), 
                 col="darkgreen", 
                 fill="navyblue", 
                 alpha=.2, binwidth=0.1) +
  ylab("Count") + 
  xlab("PTSD-PRS") +
  geom_density(col="red")

plot1

```

```{r}
#pdf("//.../hist_ptsd_prs_ancestry2.pdf", width=5.5, height=5)

x1 <- PTSD_all_data %>% group_by(PTSD_status) %>% summarise(mu1 = mean(prs, na.rm = TRUE))
plot1 <- PTSD_all_data %>% 
  filter(!is.na(prs)) %>% 
  mutate(PTSD_status = fct_relevel(PTSD_status, "Positive", "Negative")) %>% 
  ggplot(aes(x=prs, color=PTSD_status)) +
#  geom_histogram(aes(y =..density..), alpha=.2, binwidth=0.1) +
  geom_density() +
  geom_vline(data=x1, aes(xintercept=mu1, color=PTSD_status), linetype="dashed")+
  xlab("PTSD-PRS")

plot1

#dev.off()
#plot1
```




